Fix compiling with C++14 due to bit-shifting of negative integers.
See also: https://bugs.gentoo.org/show_bug.cgi?id=597364

--- a/source/core/NstCore.hpp
+++ b/source/core/NstCore.hpp
@@ -279,14 +279,14 @@
 		template<typename T>
 		inline long signed_shl(T v,uint c)
 		{
-			enum {NATIVE = T(-7) << 1 == -14};
+			enum {NATIVE = T(-(7 << 1)) == -14};
 			return Helper::ShiftSigned<T,NATIVE>::Left( v, c );
 		}
 
 		template<typename T>
 		inline long signed_shr(T v,uint c)
 		{
-			enum {NATIVE = T(-7) >> 1 == -4 || T(-7) >> 1 == -3};
+			enum {NATIVE = T(-(7 >> 1)) == -4 || T(-(7 >> 1)) == -3};
 			return Helper::ShiftSigned<T,NATIVE>::Right( v, c );
 		}
 
--- a/source/unix/gtkui/gtkui.cpp
+++ b/source/unix/gtkui/gtkui.cpp
@@ -438,7 +438,7 @@
 				GTK_DIALOG_DESTROY_WITH_PARENT,
 				GTK_MESSAGE_INFO,
 				GTK_BUTTONS_OK,
-				message);
+				"%s", message);
 	gtk_dialog_run(GTK_DIALOG(messagewindow));
 	gtk_widget_destroy(messagewindow);
 }
--- a/source/unix/gtkui/gtkui.h
+++ b/source/unix/gtkui/gtkui.h
@@ -8,7 +8,6 @@
 #elif _APPLE
 #else
 #include <gdk/gdkx.h>
-#include <gdk/gdkwayland.h>
 #endif
 
 void gtkui_init(int argc, char *argv[]);
